images/egress/dns-proxy/egress-dns-proxy.sh

```diff
@@ -15,6 +15,9 @@ IPADDR_REGEX="[[:digit:]]+\.[[:digit:]]+\.[[:digit:]]+\.[[:digit:]]+"
 PORT_REGEX="[[:digit:]]+"
 DOMAINNAME_REGEX="[[:alnum:]][[:alnum:]-]+?\.[[:alnum:].-]+"
 
+# Fetch nameservers from /etc/resolv.conf
+declare -a NAMESERVERS=$(awk '/^nameserver/ {print $2}' /etc/resolv.conf)
```

Comment:
The assignment needs parentheses in order to assign an array:
```bash
declare -a NAMESERVERS=($(awk '/^nameserver/ {print $2}' /etc/resolv.conf))
```

Reply:
It works by accident because you are not quoting `${NAMESERVERS[@]}` in that test.  Inspecting the variable using `declare -p NAMESERVERS` will show that the entire list of nameservers is in the first element.  If you use `"${NAMESERVERS[@]}"` in the loop, you will get the entire list in `$ns`.

You don't strictly need to use an array, but then it would be better not to use array syntax:
```bash
declare NAMESERVERS=$(awk '/^nameserver/ {print $2}' /etc/resolv.conf)
for ns in $NAMESERVERS; do echo $ns; done
```